### ISSUE TITLE
What is more useful Assert-Throw behavior?

### ISSUE THREAD

**Issue by nohwnd** [nohwnd/Assertions#15]
Assert throw will pass if any exception is thrown, but often we need to refine what exception we expected and what were it's properties. Which parameter is more useful to be provided __by position__? The type or the message?

```powershell
# filter on message
{ throw [InvalidOperationException]"This is wrong!" } | Assert-Throw '*wrong*'
```

```powershell
# filter on type
{ throw [InvalidOperationException]"This is wrong!" } | Assert-Throw ([InvalidOperationException])
```

**Comment by brianbunke** [comment]
I'd say string first, but no strong opinion.

**Comment by mattmcnabb** [comment]
I'd like to see both behaviors possible. You could filter on type primarily as this is similar to the pattern we use with try/catch. A second parameter should be available for filtering by message and further parameters for other exception properties.

**Comment by nohwnd** [comment]
In the end I did not make any of the paramater positional.

@mattmcnabb I just released it in [0.7.0](https://www.powershellgallery.com/packages/Assert/0.7.0). Please help me test it.
